### ISSUE TITLE
group_by + summarise result in corrupt grouped_df

### ISSUE THREAD

**Issue by jangorecki** [tidyverse/dplyr#3640]
Error does not happens every time, using 100 attempts I was able to reproduce it always, on two different machines.
```r
library(dplyr)
read.dcf(system.file("DESCRIPTION", package="dplyr"), fields="RemoteSha")[[1L]]
#[1] "379480c555c1a19af4c4bbb135662ec4cac169a6"
sessionInfo()
#R version 3.5.0 (2018-04-23)
#Platform: x86_64-pc-linux-gnu (64-bit)
#Running under: Ubuntu precise (12.04.5 LTS)
#
#Matrix products: default
#BLAS: /usr/local/lib/R/lib/libRblas.so
#LAPACK: /usr/local/lib/R/lib/libRlapack.so
#
#locale:
# [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
# [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
# [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
# [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
# [9] LC_ADDRESS=C               LC_TELEPHONE=C            
#[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
#
#attached base packages:
#[1] stats     graphics  grDevices utils     datasets  methods   base     
#
#other attached packages:
#[1] dplyr_0.7.5.9000
#
#loaded via a namespace (and not attached):
# [1] tidyselect_0.2.4 compiler_3.5.0   magrittr_1.5     assertthat_0.2.0
# [5] R6_2.2.2         pillar_1.2.3     bindrcpp_0.2.2   glue_1.2.0      
# [9] tibble_1.4.2     Rcpp_0.12.17     pkgconfig_2.0.1  rlang_0.2.0.9001
#[13] purrr_0.2.5      bindr_0.1.1     
#
set.seed(108)
DF <- data.frame(
  id1 = sample(sprintf("id%03d",1:100), 1e9, TRUE),
  v1 =  sample(5, 1e9, TRUE),
  stringsAsFactors = FALSE
)
for (i in 1:100) summarise(group_by(DF, id1), sum(v1))
#Error in summarise_impl(.data, dots) : `.data` is a corrupt grouped_df
```

**Comment by krlmlr** [comment]
This example seems to require 32 GB of RAM. Working on creating a smaller example using `gctorture2()` .

**Comment by krlmlr** [comment]
Can't replicate with the following script that requires a smaller working set:

```r
library(dplyr)
set.seed(108)
N <- 1e8
DF <- data.frame(
  id1 = sample(sprintf("id%03d",1:100), N, TRUE),
  v1 =  sample(5, N, TRUE),
  stringsAsFactors = FALSE
)

for (i in 1:100) summarise(group_by(DF, id1), sum(v1))
```

Are you seeing this only for these very large datasets?

**Comment by krlmlr** [comment]
(Tried `gctorture2(999)` too with `N <- 1e6` to no avail.)

**Comment by jangorecki** [comment]
On 1e8 I did not observe this error so far.

**Comment by krlmlr** [comment]
I do have access to a large server, will test there.

**Comment by krlmlr** [comment]
No error after 100 iterations on the server (each iteration takes ~1.5 minutes) :-(

Will run a few more.

**Comment by nickbond** [comment]
I've had a similar issue but also cannot replicate reliably.

**Comment by romainfrancois** [comment]
I think this commit fixed it. https://github.com/tidyverse/dplyr/commit/072f050038de05359aa7b9b321c1be9cfe8483b5

Can you confirm @jangorecki ?

**Comment by romainfrancois** [comment]
Probably not actually, this was a different problem

**Comment by jangorecki** [comment]
First I re-installed to latest (already after `072f050`) and could not reproduce issue. Then I checked 79d158a6b62a0c71b5a27ea67d80d6a92da4d7ea - improved error message - could not reproduce it there too. So to check if maybe reason was corrupted installation I then reverted to initial commit 379480c555c1a19af4c4bbb135662ec4cac169a6 that I was having problem with, and I was able to reproduce it there. So I checked parent of 79d158a: 8bb080c0c5a0f887d30da29a9e48c0b3f91406f1 and I was able reproduce it there.
So it seems that 79d158a6b62a0c71b5a27ea67d80d6a92da4d7ea fixed this issue.
